espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/EspressoDriver.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.drivers
+
+import android.view.View
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.action.ViewActions
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidSelectorException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.ViewFinder
+import io.appium.espressoserver.lib.model.*
+import io.appium.espressoserver.lib.viewaction.ViewGetter
+import io.appium.espressoserver.lib.viewaction.ViewTextGetter
+
+class EspressoDriver : AppDriver {
+    override val name = DriverContext.StrategyType.ESPRESSO
+
+    override fun findElement(params: Locator): BaseElement {
+        var parentView: View? = null
+        params.elementId?.let {
```

Comment:
this call could be combines with the previous one

Reply:
done